sh_scrapy/settings.py

```diff
@@ -95,12 +101,21 @@ def _load_addons(addons, settings, merged_settings, priority=0):
             raise
         skey = _get_component_base(settings, addon['type'])
         components = settings[skey]
-        path = update_classpath(addon_path)
-        components[path] = addon['order']
+        _update_component_order(components, addon_path, addon['order'])
         merged_settings.set(skey, components)
         merged_settings.setdict(addon['default_settings'], priority)
 
 
+def _merge_with_keeping_order(settings, updates):
+    for setting_key, value in updates.items():
+        if not isinstance(value, dict):
+            settings.set(setting_key, value, priority='cmdline')
+            continue
+        components = settings[setting_key]
+        for path, order in value.items():
+            _update_component_order(components, path, order)
```

Comment:
Yeah, that's how it works now: Scrapy doesn't really copy values here. Does it change anything in this specific case? We could do `setdict` with copies of the settings, but all these objects are temporary and I'm not sure if it makes any sense, unless to keep it pure and less error-prone. Other option is to drop a comment here describing this behavior for future use.